### PR TITLE
Add new error case tecOBJECT_NOT_FOUND

### PR DESCRIPTION
**Pull request by DennisDawson**
Per https://ripplelabs.atlassian.net/browse/DOC-3807, add Error Case to https://xrpl.org/nftokenacceptoffer.html.

### CODE REVIEW

**Comment by mDuo13** [comment]
The 7 broken links in the Japanese target are from #1358 and have been fixed already in master as https://github.com/XRPLF/xrpl-dev-portal/pull/1374/commits/9a09afdc69c869689a887df6c0b54e8944fa24e8 so no need to worry about those.

File: content/references/protocol-reference/transactions/transaction-types/nftokenacceptoffer.md [hunk 1]
```diff
@@ -80,7 +80,9 @@ If both offers are for the same asset, it is possible that the order in which fu
 
 In brokered mode, The offers referenced by `BuyOffer` and `SellOffer` must both specify the same `TokenID`; that is, both must be for the same `NFToken`.
 
+## Error Cases
 
+* If there is no entry with the given `SellOffer` or `BuyOffer` index on the ledger, the transaction fails with the result `tecOBJECT_NOT_FOUND`.
```
Comment (mDuo13) [inline content/references/protocol-reference/transactions/transaction-types/nftokenacceptoffer.md]: I suspect there are other error cases not noted here.

Also, please add the new error code to https://github.com/XRPLF/xrpl-dev-portal/blob/master/content/references/protocol-reference/transactions/transaction-results/tec-codes.md

**Review by mDuo13 — DISMISSED** [review]

File: content/references/protocol-reference/transactions/transaction-results/tec-codes.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ For the most part, transactions with `tec` codes take no action other than to de
 
 | Code                       | Value | Explanation                             |
 |:---------------------------|:------|:----------------------------------------|
+| `tecCANT_ACCEPT_OWN_NFTOKEN_OFFER` | 157 | The transaction tried to accept an offer that was placed by the same account to buy or sell a [non-fungible token](non-fungible-tokens.html). _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tec-codes.md:18]: :+1:

File: content/references/protocol-reference/transactions/transaction-results/tec-codes.md [hunk 3]
```diff
@@ -27,11 +28,16 @@ For the most part, transactions with `tec` codes take no action other than to de
...
 | `tecINSUF_RESERVE_OFFER`   | 123   | The transaction failed because the sending account does not have enough XRP to create a new Offer. (See: [Reserves](reserves.html)) |
 | `tecINSUFF_FEE`            | 136   | The transaction failed because the sending account does not have enough XRP to pay the [transaction cost](transaction-cost.html) that it specified. (In this case, the transaction processing destroys all of the sender's XRP even though that amount is lower than the specified transaction cost.) This result only occurs if the account's balance decreases _after_ this transaction has been distributed to enough of the network to be included in a consensus set. Otherwise, the transaction fails with [`terINSUF_FEE_B`](ter-codes.html) before being distributed. |
+| `tecINSUFFICIENT_FUNDS`    | 158 | One of the accounts involved does not hold enough of a necessary asset. _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tec-codes.md:31]: :+1:

File: content/references/protocol-reference/transactions/transaction-results/tec-codes.md [hunk 4]
```diff
@@ -27,11 +28,16 @@ For the most part, transactions with `tec` codes take no action other than to de
...
+| `tecINSUFFICIENT_PAYMENT`  | 161 | The amount specified is not enough to pay all fees involved in the transaction. For example, when trading a non-fungible token, the buy amount may not be enough to pay both the broker fee and the sell amount. _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tec-codes.md:32]: :+1:

File: content/references/protocol-reference/transactions/transaction-results/tec-codes.md [hunk 5]
```diff
@@ -27,11 +28,16 @@ For the most part, transactions with `tec` codes take no action other than to de
...
 | `tecINTERNAL`              | 144   | Unspecified internal error, with transaction cost applied. This error code should not normally be returned. If you can reproduce this error, please [report an issue](https://github.com/ripple/rippled/issues). |
 | `tecINVARIANT_FAILED`      | 147   | An invariant check failed when trying to execute this transaction. Added by the [EnforceInvariants amendment][]. If you can reproduce this error, please [report an issue](https://github.com/ripple/rippled/issues). |
 | `tecKILLED`                | 150   | The [OfferCreate transaction][] specified the `tfFillOrKill` flag and could not be filled, so it was killed. _(Added by the [fix1578 amendment][].)_ |
+| `tecMAX_SEQUENCE_REACHED`  | 153   | A sequence number field is already at its maximum. This includes the `MintedNFTokens` field. _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tec-codes.md:37]: :+1:

File: content/references/protocol-reference/transactions/transaction-results/tec-codes.md [hunk 6]
```diff
@@ -27,11 +28,16 @@ For the most part, transactions with `tec` codes take no action other than to de
...
 | `tecNEED_MASTER_KEY`       | 142   | This transaction tried to cause changes that require the master key, such as [disabling the master key or giving up the ability to freeze balances](accountset.html#accountset-flags). [New in: rippled 0.28.0][] |
+| `tecNFTOKEN_BUY_SELL_MISMATCH` | 155 | The [NFTokenAcceptOffer transaction][] :not_enabled: attempted to match incompatible offers to buy and sell a non-fungible token. _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tec-codes.md:39]: :+1:

File: content/references/protocol-reference/transactions/transaction-results/tec-codes.md [hunk 7]
```diff
@@ -27,11 +28,16 @@ For the most part, transactions with `tec` codes take no action other than to de
...
+| `tecNFTOKEN_OFFER_TYPE_MISMATCH` | 156 | One or more of the offers specified in the transaction was not the right type of offer. (For example, a buy offer was specified in the `NFTokenSellOffer` field.) _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tec-codes.md:40]: :+1:

File: content/references/protocol-reference/transactions/transaction-results/tec-codes.md [hunk 8]
```diff
@@ -43,7 +49,9 @@ For the most part, transactions with `tec` codes take no action other than to de
 | `tecNO_LINE_REDUNDANT`     | 127   | The transaction failed because it tried to set a trust line to its default state, but the trust line did not exist. |
 | `tecNO_PERMISSION`         | 139   | The sender does not have permission to do this operation. For example, the [EscrowFinish transaction][] tried to release a held payment before its `FinishAfter` time, someone tried to use [PaymentChannelFund][] on a channel the sender does not own, or a [Payment][] tried to deliver funds to an account with the "DepositAuth" flag enabled. |
 | `tecNO_REGULAR_KEY`        | 131   | The [AccountSet transaction][] tried to disable the master key, but the account does not have another way to [authorize transactions](transaction-basics.html#authorizing-transactions). If [multi-signing](multi-signing.html) is enabled, this code is deprecated and `tecNO_ALTERNATIVE_KEY` is used instead. |
+| `tecNO_SUITABLE_NFTOKEN_PAGE` | 154 | ***TODO*** |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tec-codes.md]: The account is attempting to mint or acquire an `NFTokenID` that cannot be held by any of the account's `NFToken` directory pages.  This circumstance can be engineered with some effort, but should never occur through simple minting and/or acquisition of `NFTokens`.

File: content/references/protocol-reference/transactions/transaction-results/tec-codes.md [hunk 9]
```diff
@@ -43,7 +49,9 @@ For the most part, transactions with `tec` codes take no action other than to de
...
 | `tecNO_TARGET`             | 138   | The transaction referenced an Escrow or PayChannel ledger object that doesn't exist, either because it never existed or it has already been deleted. (For example, another [EscrowFinish transaction][] has already executed the held payment.) Alternatively, the destination account has `asfDisallowXRP` set so it cannot be the destination of this [PaymentChannelCreate][] or [EscrowCreate][] transaction. |
+| `tecOBJECT_NOT_FOUND`      | 160   | One of the objects specified by this transaction did not exist in the ledger. _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tec-codes.md:54]: :+1:

File: content/references/protocol-reference/transactions/transaction-results/tef-codes.md [hunk 10]
```diff
@@ -28,6 +28,7 @@ These codes indicate that the transaction failed and was not included in a ledge
 | `tefINVARIANT_FAILED`   | An invariant check failed when trying to claim the [transaction cost](transaction-cost.html). Added by the [EnforceInvariants amendment][]. If you can reproduce this error, please [report an issue](https://github.com/ripple/rippled/issues). |
 | `tefMASTER_DISABLED`    | The transaction was signed with the account's master key, but the account has the `lsfDisableMaster` field set. |
 | `tefMAX_LEDGER`         | The transaction included a [`LastLedgerSequence`](reliable-transaction-submission.html#lastledgersequence) parameter, but the current ledger's sequence number is already higher than the specified value. |
+| `tefNFTOKEN_IS_NOT_TRANSFERABLE` | The transaction attempted to send a [non-fungible token](non-fungible-tokens.html) to another account, but the `NFToken` has the `lsfTransferable` flag disabled and the transfer would not be to or from the issuer. _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tef-codes.md:31]: :+1:

File: content/references/protocol-reference/transactions/transaction-results/tem-codes.md [hunk 11]
```diff
@@ -20,6 +20,7 @@ These codes indicate that the transaction was malformed, and cannot succeed acco
 | `temBAD_FEE`                  | The transaction improperly specified its `Fee` value, for example by listing a non-XRP currency or some negative amount of XRP. |
 | `temBAD_ISSUER`               | The transaction improperly specified the `issuer` field of some currency included in the request. |
 | `temBAD_LIMIT`                | The [TrustSet transaction][] improperly specified the `LimitAmount` value of a trust line. |
+| `temBAD_NFTOKEN_TRANSFER_FEE` | The [NFTokenMint transaction][] improperly specified the `TransferFee` field of the transaction. _(Added by the [NonFungibleTokensV1 amendment][]. :not_enabled:)_ |
```
Comment (scottschurr) [inline content/references/protocol-reference/transactions/transaction-results/tem-codes.md:23]: :+1:

**Review by scottschurr — APPROVED** [review]
The added `TER` code descriptions look good to me.  I also left a first approximation for what can be said regarding `techno_SUITABLENFTOKEN_PAGE`.  I can provide more background on that if you would like.

**Comment by mDuo13** [comment]
- Added the missing description @scottschurr pointed out
- Fixed a couple places where `"Fee"` was specified as an int instead of a string (in these docs & the reliable transaction submission tutorial)
- Rebased on master and resolved merge commits
- Corrected a statement about URI being in uppercase hex and added a note about the bug in xrpl.js

**Comment by mDuo13** [comment]
The broken links to the known amendments page are expected (I plan to update the known amendments page in a separate PR)

**Comment by mDuo13** [comment]
#1409 fixes the broken links to the amendment description

**Pull request merged**